src/System.Diagnostics.Process/src/System/Collections/Specialized/StringDictionaryWithComparer.cs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System;
+using System.Globalization;
+using System.Collections.Generic;
+
+namespace System.Collections.Specialized {
+    
+    /* This class is an internal class used by System.Diagnostics.Proccess on property EnvironmentVariables which returns an StringDictionary. Since we need an OrdinalIgnoreCase
+    comparer for non casesensitive environments and Ordinal comparer for case sensitive environments, we need this workaround in order to use StringDictionary. We should 
+    override StringDictionary's method and use our own Dictionary<string,string> with comparer since this class is internal and StringDictionary _contents Hashtable 
+    is private and we don't want to make that field protected. */
+
+    [Serializable]
+    internal class StringDictionaryWithComparer : StringDictionary {
+
+        private Dictionary<string, string> _contents;
+        private readonly object _syncRoot = new Object();
+
+        public StringDictionaryWithComparer() : this(StringComparer.OrdinalIgnoreCase) { }
+
+        public StringDictionaryWithComparer(IEqualityComparer<string> comparer)
+        {
+            _contents = new Dictionary<string, string>(comparer);         
+        }
+
+        public override int Count
+        {
+            get
+            {
+                return _contents.Count;
+            }
+        }
+        
+        public override string this[string key]
+        {
+            get
+            {
+                if(key == null)
+                {
+                    throw new ArgumentNullException(nameof(key));
+                }
+
+                return _contents[key];
+            }
+            set 
+            {
+                if(key == null)
+                {
+                    throw new ArgumentNullException(nameof(key));
+                }
+
+                _contents[key] = value;
+            }
+        }
+
+        public override ICollection Keys
```

Comment:
Nit: these one-liners can be simplified with C# 6 express-bodied members, e.g.
```C#
public override ICollection Keys => _contents.Keys;

public override ICollection Values => _contents.Values;

public override bool IsSynchronized => false;
```
Etc.